romann-broque/Minishell
# Mve4 cwd (#31)

* Mve3 us1 b UI ltin (#24)

* update yml file

* tracker set

* builtin implementation

* clean some array declarations

* valgrind tests in progress

* exec_builtin file

* prompt dynamic allocations are managed by the free_manager function

* add free_manager in exit_shell function

* valgrind tests

* modified yml tests

* modified yml tests (valgrind installing)

* valgrind docker yml tests

* valgrind docker yml tests

* valgrind docker yml tests 3

* valgrind docker yml tests 4

* valgrind docker yml tests 5

* categorize tests

* categorize tests 2

* categorize tests 3

* clean Makefile and free_manager

* valgrind errors are printed when CUNIT tests runs

* compiler is now cc + clean valgrind tests

* Mve3 us2 binary st1 path (#25)

* finished the paths functions, still has to be tested tho

* finished testing path functions

* corrected the parser

* corrected empty path case

* removed small print

* applied comments

* applied last comment

* last change

* execution implementation + some min tests removed to match with the new minishell implementation

* fix double free with the path

* finished CWD and started the batch

* handled spec cd arg

* cd in progress

* environment dup in progress

* protected PWD and added a test case

* global variable + ft_getenv in progress

* tests updated + remove leaks from env at exit

* ft_getenv tests

* tests for change_var function

* update cd and pwd to support env -i

* tests updated

* update change_var function

* fix syntax and parser errors

* manage remove file

* just add test

* init_shell + check_pos

* cd changed

* init env with PWD variable

* add CDPATH feature for cd builtin

* cd : symbolic links in progress

* symbolic links resolution in progress

* symbolic links resolved !

* clean code

* add some min tests

* clean code again

* cd error arg

* change tests to check git actions

* fsanitize tests for git actions first

* valgrind tests first

* just change to printf

* comment CUNIT tests and print to debug

* protect change_var against NULL var_value

* debug update_cwd_var

* debug in progress

* rm_backpath debug

* rm_backpath debug 2

* alleluyah

* cd now supports symbolic links

* ignore norm during git actions

* remove useless recompilations

* rm_double_slash is back

* silent end slashj

* discover new issue: empty arguments are removed :o

* fix empty arguments for commands

* add tests

* fix tests

* fix error message with command not found

* apply comments

---------
